src/middlewared/middlewared/etc_files/local/apache24/Includes/webdav.conf

```diff
@@ -97,9 +97,13 @@ Listen ${webdav_config['tcpport']}
 			AllowMethods GET OPTIONS PROPFIND
 		</Location>
 	% endif
-<% if share['perm']:
-	change_perms(share['path'])	
-%>
+<%
+	if not os.path.exists(share['path']):
+		os.makedirs(share['path'])
```

Comment:
Not sure if we want to create the dir in the config file gen, wouldnt it be enough in share create?

Reply:
Actually i was confused about that. So that will only work if the user is updating/creating the share - this is for compatibility for old shares. It can be removed if you think this doesn't make much sense here. Thoughts ?

Reply:
I am not sure other shares do it that way?

We either mkdir on create/update or config gen, not both. I vote for create/update.

Reply:
I agree. Will have the PR updated asap ; )